README.md

```diff
@@ -1,16 +1,611 @@
+
+### 3.1.6 Signature
+
+The `sig` field MUST contain the signature of the other fields. To construct the payload, the other fields MUST first be canonicalized as `dag-cbor`.
+
+If serialized as JSON, the `sig` field MUST be serialized as [unpadded base64url](https://datatracker.ietf.org/doc/html/rfc4648#section-5).
+
+## 3.2 IPLD Schema
+
+``` ipldsch
+type Invocation struct {
+  inv &UCAN  -- The UCAN providing authority
+  v   SemVer -- Version
+  run Scope  -- Which actions to invoke
+  nnc String -- Nonce
+  ext Any    -- Extended fields
+  sig Bytes  -- Signature
+}
+
+type Scope enum {
+  | All ("*")
+  | Action
+}
+
+type Action enum {
+  using  URI 
+  do     Ability
+  inputs Any
+} 
+```
+
+## 3.3 JSON Examples
+
+### 3.3.1 Run All
+
+ ``` js
+{
+  "ucan/invoke": "QmYW8Z58V1v8R25USVPUuFHtU7nGouApdGTk3vRPXmVHPR",
+  "v": "0.1.0",
+  "nnc": "abcdef",
+  "ext": null,
+  "sig": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA"
+}
+```
+
+ ``` js
+{
+  "ucan/invoke": "QmYW8Z58V1v8R25USVPUuFHtU7nGouApdGTk3vRPXmVHPR",
+  "v": "0.1.0",
+  "nnc": "abcdef",
+  "run": "*", // Exmplicitely "run all"
+  "ext": null,
+  "sig": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA"
+}
+```
+
+### 3.3.2 Pipelines
+
+The following examples both express the following dataflow graph:
+
+```
+              ┌────────────────────────────┐
+              │                            │
+              │ dns://example.com?TYPE=TXT │
+              │        crud/update         │
+              │                            │
+              └───────┬──────────┬─────────┘
+                      │          │
+                      │          │
+┌─────────────────────▼───┐  ┌───▼────────────────────────┐
+│                         │  │                            │
+│ mailto:alice@exaple.com │  │ mailto://alice@example.com │
+│         msg/send        │  │          msg/send          │
+│     bob@example.com     │  │      carol@exmaple.com     │
+│                         │  │                            │
+└─────────────────────┬───┘  └───┬────────────────────────┘
+                      │          │
+                      │          │
+             ┌────────▼──────────▼────────┐
+             │                            │
+             │ https://example.com/events │
+             │         crud/create        │
+             │                            │
+             └────────────────────────────┘
+```
+
+#### 3.3.2.1 Batched 
+
+ ``` json
+{
+  "ucan/invoke": "QmYW8Z58V1v8R25USVPUuFHtU7nGouApdGTk3vRPXmVHPR",
+  "v": "0.1.0",
+  "nnc": "abcdef",
+  "ext": null,
+  "run": {
+    "update-dns" : {
+      "using": "dns://example.com?TYPE=TXT":
+      "do": "crud/update",
+      "inputs": { 
+         "value": "hello world",
+         "retries": 5
```

Comment:
I'm not exactly sure where I fall, but `retries` feels a little out of place in this spec, compared to everything else that it tries to cover, since it's a non-functional concern of the action rather than a description of what's being delegated.

It's definitely an important concern, and it probably belongs somewhere, but if it belongs here, then what about closely related ideas, like the backoff policy for the action when performing retries? There's a big difference between rapidly hitting a down service 5 times in quick succession, versus doing so 5 times with exponential backoff between each attempt. That kind of backoff strategy feels more obviously out of place in this spec, but since it's so conceptually similar to `retries`, it has me questioning that field too.

Reply:
@QuinnWilton this part of the spec is open fields for the caller to define. If the executor needs retries as input, then it needs to be in this field.